Experiment: Test Combining

korourke5 opened the pull request:
# CRCsim PR Template

## Description
<!-- Provide a brief description of the changes in this PR -->
Created new branch for test combining experiment combining FIT and Blood tests in serial and parallel. Changes:
- Parameters.py: ran combine_tests.py and added these new parameters here.
- Prepare.py: added new parameters in to scenarios.append()

## PR Type
<!-- Mark the appropriate option with an [x] (no spaces around the x) -->
- [ ] Feature (intended to merge)
- [X] Experiment (not intended to merge)

## Experiment review checklist
<!-- Make sure all items are checked before submitting the PR -->
- [X] This experiment branch has the `exp-` prefix
- [X] This experiment uses the appropriate `crcsim` commit hash (typically, the latest commit in the `main` branch). This ensures that the experiment uses the latest version of the model. Note: used experiment-template
- [X] This experiment uses the correct baseline `parameters.json` file
    - [X] Correct screening start and end ages for the experiment
    - [X] Latest values of all calibrated parameters
    - [X] All necessary tests and latest test parameters
    - [X] Latest cost parameters
- [X] My `prepare.py` script applies incidence rate ratio (IRR) adjustment if appropriate for this experiment
- [X] I have run `prepare.py` and:
    - [X] Confirmed that all scenarios necessary for this experiment were created
    - [X] Confirmed that no extraneous scenarios are in the `scenarios/` directory (eg, uncommitted local changes from a previous experiment)
    - [X] Spot-checked a sample of scenario parameter files to ensure that my scenario creation logic works as expected
- [X] I have updated `crcsim/experiment/README.md` with detailed information about the experiment's goals, scenarios, and corresponding AWS objects
- [X] I am opening this PR as a **Draft** PR

## Experiment review process
<!-- No checkboxes here, because this happens after you submit the PR -->
*All experiments must follow each step of this review process.*
1. Contributor checks all items in experiment review checklist
1. Contributor opens **Draft PR** and tags a collaborator to review
1. Reviewer conducts a thorough review, including pulling experiment branch, running `prepare.py`, and spot-checking scenarios
1. Contributor and reviewer address any issues identified during review
1. Reviewer explicitly approves PR
1. Contributor builds image and pushes to ECR, copies experiment files to S3, and runs experiment
1. Contributor analyzes experiment (eg `crcsim/experiment/summarize.py`)
1. Contributor pushes results summary, so we have documentation of detailed results (eg `crcsim/experiment/summary/summarized.xlsx`)
1. Contributor updates experiment README with a detailed summary of results
1. Contributor closes this PR with an informative comment (eg, brief summary of results)

**Note that experiment PRs are never merged into `main`!** The PR is closed, and the experiment is maintained as a separate branch. That's why we keep all experiment PRs as drafts.

korourke5 committed:
Update parameters.json to add blood test parameters

korourke5 committed:
Edit parameters.json and prepare.py to include
combined blood and FIT tests

korourke5 committed:
Added the scenarios created and updated readme

sandypreiss commented on crcsim/experiment/prepare.py:
```diff
@@ -137,6 +140,24 @@ def create_scenarios() -> List:
             .transform(transform_lesion_risk_alpha(1.19))
         )
 
+        scenarios.append(
+            Scenario(name=f"Blood_{scenario}", params=get_default_params())
+            .transform(transform_initial_compliance(screening_rate))
+            .transform(transform_lesion_risk_alpha(1.19))
+        )
+
+        scenarios.append(
+            Scenario(name=f"FIT_Blood_TestCombiningMethod.PARALLEL_{scenario}", params=get_default_params())
+            .transform(transform_initial_compliance(screening_rate))
+            .transform(transform_lesion_risk_alpha(1.19))
+        ) 
+        
+        scenarios.append(
+            Scenario(name=f"FIT_Blood_TestCombiningMethod.SERIAL_{scenario}", params=get_default_params())
+            .transform(transform_initial_compliance(screening_rate))
+            .transform(transform_lesion_risk_alpha(1.19))
+        )
```
After we modify the test combining script to set the proportion for combined tests to 0, the base params will assign FIT to all agents. So these tests also need to use `transform_routine_test_proportion` to set the proportion for the relevant test to 1 and all others to 0 (same pattern as Colonoscopy).

sandypreiss commented:
@korourke5 I pushed my changes to test combining to `experiment-template`. You can merge that branch into yours, respond to my other comment, and rerun `prepare.py`. That should get you ready to run the experiment.

korourke5 committed:
Merge remote-tracking branch 'origin/experiment-
template' into exp-test-combining

korourke5 committed:
Updated prepare.py and parameters.json to include
the most recent versions of FIT_Blood_parallel and
FIT_Blood_serial from updated combine_tests.py

korourke5 committed:
Added results in summary, updated the S3 bin

korourke5 commented:
@sandypreiss I have updated with the latest run. I wasn't sure exactly which results we were looking for so I have not yet updated the README.

sandypreiss commented:
Bottom line is that the results look as they ought to with the combined test. 

FIT + blood in parallel leads to less CRC and more diagnostic tests (ie, positive screenings) than either test alone. FIT + blood in serial leads to more CRC and fewer diagnostic tests than either test alone.

You can copy that exactly for the readme. Also, please delete the `~` prefixed version of the summary sheet.

korourke5 committed:
Update README.md with results.

korourke5 committed:
Delete crcsim/experiment/summary/~$summarized.xlsx

korourke5 commented:
@sandypreiss This PR should be all set for closing out.
